src/transport.rs

```diff
@@ -50,66 +52,101 @@ impl WireNetwork {
                 .await
                 .unwrap_or_else(|op| error!("Error in listen_out {:?}", op));
         });
+
+        let (dec_chan_tx, dec_chan_rx) = mpsc::channel(channel_size);
+
+        let c = conf.clone();
```

Comment:
I suspect some `clone` are not needed, what's about `clippy`?

Reply:
Removed one of the clone calls.